### PR TITLE
refactored HomePagePositionsContainer to use Hooks/React 16. Tried to…

### PR DESCRIPTION
**Pull request by elizabeth-jimenez**
… clean up homepage positions logic. homepage now showing featured positions

Dual Merge: [API](https://github.com/MetaPhase-Consulting/State-TalentMAP-API/pull/399) <-- update:  merged
😓 😫 

**Featured Positions**
- [x] What happens when {} data returned on first call: FE successfully calls next fallback
- [x] What happens when {} data returned on first call and then 2nd call returns !{} data: FE successfully receives next data set and updates title accordingly.
- [x] What happens when {} data returned on 1st and 2nd call: FE successfully calls next fallback
- [x] What happens when {} data returned on 1st and 2nd call and 3rd call returns !{} data: FE successfully receives next data set and updates title accordingly.
- [x] What happens when {} data returned on 1st, 2nd, and 3rd call for: FE shows nothing for Featured Positions

**Recommended Positions**
- [x] What happens when results.count is 0 returned on first call: FE successfully calls next fallback
- [x] What happens when results.count is 0 returned on first call and then 2nd call returns results.count is !0:  FE successfully receives next data set and updates title accordingly.
- [x] What happens when results.count is 0 returned on 1st and 2nd call: FE successfully calls next fallback
- [x] What happens when results.count is 0 returned on 1st and 2nd call and 3rd call returns results.count is !0: FE successfully receives next data set and updates title accordingly.
- [x] What happens when results.count is 0  returned on 1st, 2nd, and 3rd call: FE successfully shows the error message alerting the user that they have no current AP Favorites

### CODE REVIEW

File: src/actions/homePagePositions.js [hunk 1]
```diff
@@ -1,102 +1,185 @@
...
+
+
+// export const HIGHLIGHTED_POSITIONS_QUERY = () => '/available_position/highlight/?limit=3';
+// export const GET_SKILL_CODE_POSITIONS_QUERY = skillCodes =>
+// `/fsbid/available_positions/?position__skill__in=${skillCodes}&limit=3`;
+// export const GET_GRADE_POSITIONS_QUERY = grade =>
+// `/fsbid/available_positions/?position__grade__code
+// __in=${grade}&limit=3&ordering=-${COMMON_PROPERTIES.posted}`;
+// export const RECENTLY_POSTED_POSITIONS_QUERY = () =>
+// `/fsbid/available_positions/?limit=3&ordering=-${COMMON_PROPERTIES.posted}`;
```
Comment (mjoyce91) [inline src/actions/homePagePositions.js:34]: needed?

**Comment by mjoyce91** [comment]
The button state for Favorite doesn't change after click for these

Screen Shot 2020-08-17 at 1 39 13 PM

**Comment by elizabeth-jimenez** [comment]
> The button state for Favorite doesn't change after click for these
> 
> Screen Shot 2020-08-17 at 1 39 13 PM

😭

File: src/actions/homePagePositions.js (hunk 1 above)
Comment (elizabeth-jimenez) [reply src/actions/homePagePositions.js:34]: you never know.... 😝 
Will remove!

**Comment by elizabeth-jimenez** [comment]
> The button state for Favorite doesn't change after click for these
> 
> Screen Shot 2020-08-17 at 1 39 13 PM

lol i tried to "clean" the homepage logic too much and removed that functionality lol!

**Comment by elizabeth-jimenez** [comment]
updating last snapshots and tests

**Comment by mjoyce91** [comment]
I wonder if we should make it a condition that at least two or three positions appear. Seems weird to just have one.

Screen Shot 2020-08-24 at 9 51 12 AM

File: src/Components/HomePagePositions/HomePagePositions.test.jsx [hunk 2]
```diff
@@ -43,72 +51,109 @@ describe('HomePageComponent', () => {
...
+  it('displays proper title and icon when favoritedPositions displayed', () => {
+    const wrapper = shallow(<HomePagePositions
+      {...props}
+      homePageRecommendedPositions={{ ...props.homePageRecommendedPositions,
+        name: 'favoritedPositions',
+      }}
+    />);
+    expect(wrapper.find('HomePagePositionsSection').at(1).prop('title')).toBe('Favorited Positions');
+    expect(wrapper.find('HomePagePositionsSection').at(1).prop('icon')).toBe('star');
+    expect(wrapper.find('HomePagePositionsSection')).toHaveLength(2);
```
Comment (mjoyce91) [inline src/Components/HomePagePositions/HomePagePositions.test.jsx:119]: 👍

File: src/Components/HomePagePositions/HomePagePositions.test.jsx [hunk 3]
```diff
@@ -3,16 +3,24 @@ import toJSON from 'enzyme-to-json';
 import React from 'react';
```
Comment (mjoyce91) [inline src/Components/HomePagePositions/HomePagePositions.test.jsx:3]: good tests here!

**Comment by elizabeth-jimenez** [comment]
> I wonder if we should make it a condition that at least two or three positions appear. Seems weird to just have one.
> 
> Screen Shot 2020-08-24 at 9 51 12 AM

I agree, and hopefully that'll be rare in Production, but for locally, some Personas only get 2 pages of results, so we wouldn't be able to show the randomizing feature. The easiest fix is to limit the randomization to not include the last page of results, but idk if that's wise?

File: src/Components/HomePagePositions/HomePagePositions.test.jsx (hunk 2 above)
Comment (elizabeth-jimenez) [reply src/Components/HomePagePositions/HomePagePositions.test.jsx:119]: :)

**Comment by mjoyce91** [comment]
@elizabeth-jimenez yeah good point. I think this is fine for now. Just something to think about

**Comment by elizabeth-jimenez** [comment]
🍰 🎊 Yay!

**Pull request merged**